src/custom/Feedback/FeedbackButton.tsx

```diff
@@ -0,0 +1,246 @@
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [submitted, setSubmitted] = useState<boolean>(false);
+  const [category, setCategory] = useState<FeedbackDataItem | undefined>();
+  const [messageValue, setMessageValue] = useState<string | undefined>();
+  const feedbackTextRef = useRef<HTMLTextAreaElement>(null);
+  const [isChecked, setIsChecked] = useState<boolean>(false);
+
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setIsChecked(event.target.checked);
+  };
+
+  const handleFeedback = () => {
+    setIsOpen(!isOpen);
+    setCategory(feedbackData[0]);
+    setSubmitted(false);
+    setMessageValue('');
+  };
+
+  const handleSubmit = () => {
+    setSubmitted(true);
+    if (messageValue && isChecked) {
+      onSubmit({ label: category?.label || '', message: messageValue });
+    }
+    setTimeout(() => {
+      setIsOpen(false);
+      setCategory(undefined);
+      setSubmitted(false);
+      setMessageValue('');
+    }, 2000);
+  };
+
+  return (
+    <Container isOpen={isOpen}>
+      {submitted ? (
+        <FeedbackMessage isOpen={isOpen}>
+          <SuccessIcon width={'32'} height={'32'} />
+          We got your concern. Thank you!
+        </FeedbackMessage>
+      ) : (
+        <>
+          <FeedbackButton onClick={handleFeedback}>Feedback</FeedbackButton>
+
+          <ModalCard
+            onClose={() => {}}
+            open={true}
+            closeComponent={
+              <CloseButton onClick={() => setIsOpen(false)}>
+                <CloseIcon />
+              </CloseButton>
+            }
+            actions={
+              <div
+                style={{
+                  display: 'flex',
+                  alignItems: 'center'
+                }}
+              >
+                <div style={{ display: 'flex', alignItems: 'center' }}>
+                  <FormControlLabel
+                    label={''}
+                    style={{
+                      color: 'white',
+                      fontSize: '2px',
+                      marginLeft: '0px',
+                      marginRight: '5px'
+                    }}
+                    control={
+                      <Checkbox
+                        style={{ color: '#fff' }}
+                        checked={isChecked}
+                        onChange={handleCheckboxChange}
+                      />
+                    }
+                  />
+                  <Typography style={{ fontSize: '0.8rem', marginRight: '10px', color: 'white' }}>
+                    We may email you for more information or updates
+                  </Typography>
+                </div>
+                <FeedbackSubmitButton
+                  type="submit"
+                  disabled={!(messageValue && isChecked)}
+                  isOpen={!(messageValue && isChecked)}
+                  onClick={handleSubmit}
+                >
+                  Send
```

Comment:
Fix this, please. My mockup and my feedback both have not been implemented here.